### PR TITLE
python312Packages.pyopencl: 2024.2.6 -> 2024.2.7

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

- **python312Packages.pyopencl: 2024.2.6 -> 2024.2.7**
- **python312Packages.pyopencl: add GaetanLepage as maintainer**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
I will need a bit of help on this one as I get weird `ModuleNotFound` errors:
```
Check whether the following modules can be imported: pyopencl pyopencl.array pyopencl.compyte
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/04gg5w1s662l329a8kh9xcwyp0k64v5a-python3-3.12.4/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/nix/store/ymsdh7v94cpj4p0ncakir43lkx9hs051-python3.12-pyopencl-2024.2.7/lib/python3.12/site-packages/pyopencl/__init__.py", line 29, in <module>
    import pyopencl.cltypes  # noqa: F401
    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ymsdh7v94cpj4p0ncakir43lkx9hs051-python3.12-pyopencl-2024.2.7/lib/python3.12/site-packages/pyopencl/cltypes.py", line 25, in <module>
    from pyopencl.tools import get_or_register_dtype
  File "/nix/store/ymsdh7v94cpj4p0ncakir43lkx9hs051-python3.12-pyopencl-2024.2.7/lib/python3.12/site-packages/pyopencl/tools.py", line 139, in <module>
    from pyopencl.compyte.dtypes import TypeNameNotKnown  # noqa: F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'pyopencl.compyte'
```

File: pkgs/development/python-modules/pyopencl/default.nix [hunk 1]
```diff
@@ -60,32 +64,29 @@ buildPythonPackage rec {
...
-  doCheck = false;
-
-  pythonImportsCheck = [ "pyopencl" ];
+  pythonImportsCheck = [
+    "pyopencl"
+    "pyopencl.array"
+    "pyopencl.compyte"
+  ];
 
-  meta = with lib; {
```
Comment (doronbehar) [inline pkgs/development/python-modules/pyopencl/default.nix:86]: I'd put that in a separate commit.

File: pkgs/development/python-modules/pyopencl/default.nix [hunk 2]
```diff
@@ -60,32 +64,29 @@ buildPythonPackage rec {
...
     platformdirs
     pytools
-    six
   ];
 
   nativeCheckInputs = [ pytestCheckHook ];
 
   preBuild = ''
     export HOME=$(mktemp -d)
-    rm -rf pyopencl
```
Comment (doronbehar) [inline pkgs/development/python-modules/pyopencl/default.nix:78]: This feels related to https://github.com/NixOS/nixpkgs/issues/255262 . Try to `cd $out` in `preCheck` instead, and please link to the said issue in a comment.

File: pkgs/development/python-modules/pyopencl/default.nix [hunk 3]
```diff
@@ -1,54 +1,58 @@
...
   pytools,
   six,
+
+  # checks
+  pytestCheckHook,
 }:
 
 let
-  os-specific-buildInputs = if stdenv.isDarwin then [ darwin.apple_sdk.frameworks.OpenCL ] else [ ocl-icd ];
+  os-specific-buildInputs =
```
Comment (doronbehar) [inline pkgs/development/python-modules/pyopencl/default.nix:36]: I'd put that in the same separate commit as well.

File: pkgs/development/python-modules/pyopencl/default.nix [hunk 4]
```diff
@@ -1,54 +1,58 @@
...
-  oldest-supported-numpy,
-  opencl-headers,
   platformdirs,
-  pybind11,
-  pytestCheckHook,
   pytools,
   six,
+
+  # checks
+  pytestCheckHook,
```
Comment (doronbehar) [inline pkgs/development/python-modules/pyopencl/default.nix:32]: Same.

File: pkgs/development/python-modules/pyopencl/default.nix [hunk 5]
```diff
@@ -1,54 +1,58 @@
...
 
   # build-system
   cmake,
-  scikit-build-core,
-  pathspec,
-  ninja,
   nanobind,
+  ninja,
+  numpy,
+  oldest-supported-numpy,
```
Comment (doronbehar) [inline pkgs/development/python-modules/pyopencl/default.nix:14]: Can be removed? I'd recommend using `nixd` to detect unused inputs.

File: pkgs/development/python-modules/pyopencl/default.nix [hunk 6]
```diff
@@ -60,32 +64,29 @@ buildPythonPackage rec {
     pybind11
   ] ++ os-specific-buildInputs;
 
-  propagatedBuildInputs = [
-    appdirs
-    cffi
-    decorator
-    mako
+  dependencies = [
     numpy
```
Comment (doronbehar) [inline pkgs/development/python-modules/pyopencl/default.nix:68]: Speaking of `numpy`, could it be this package supports both numpy versions? Perhaps this shouldn't be propagated.

File: pkgs/development/python-modules/pyopencl/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pyopencl/default.nix:78]: It also explains the error you receive

**Comment by GaetanLepage** [comment]
Ok, I will leave the test weirdness out for now.
I have only left the update and the cleaning bits.
Also, everything has been split in smaller commits.
The package builds fine now.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337876` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.loopy</li>
    <li>python311Packages.loopy.dist</li>
    <li>python311Packages.pyopencl</li>
    <li>python311Packages.pyopencl.dist</li>
    <li>python311Packages.reikna</li>
    <li>python311Packages.reikna.dist</li>
    <li>python311Packages.sasmodels</li>
    <li>python311Packages.sasmodels.dist</li>
    <li>python312Packages.loopy</li>
    <li>python312Packages.loopy.dist</li>
    <li>python312Packages.pyopencl</li>
    <li>python312Packages.pyopencl.dist</li>
    <li>python312Packages.reikna</li>
    <li>python312Packages.reikna.dist</li>
    <li>python312Packages.sasmodels</li>
    <li>python312Packages.sasmodels.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337876` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337876` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python311Packages.loopy</li>
    <li>python311Packages.loopy.dist</li>
    <li>python311Packages.pyopencl</li>
    <li>python311Packages.pyopencl.dist</li>
    <li>python311Packages.reikna</li>
    <li>python311Packages.reikna.dist</li>
    <li>python311Packages.sasmodels</li>
    <li>python311Packages.sasmodels.dist</li>
    <li>python312Packages.loopy</li>
    <li>python312Packages.loopy.dist</li>
    <li>python312Packages.pyopencl</li>
    <li>python312Packages.pyopencl.dist</li>
    <li>python312Packages.reikna</li>
    <li>python312Packages.reikna.dist</li>
    <li>python312Packages.sasmodels</li>
    <li>python312Packages.sasmodels.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyopencl/default.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pyopencl/default.nix:62]: a topic for another PR

**Review by SuperSandro2000 — APPROVED** [review]
we don't need to do micro commits while also not doing everything in one. I think the middle ground in between is the right choice but we leave it like it is now.

File: pkgs/development/python-modules/pyopencl/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pyopencl/default.nix]: or statix

**Comment by dotlambda** [comment]
Excuse me?? https://github.com/NixOS/nixpkgs/pull/336885

**Comment by GaetanLepage** [comment]
> Excuse me?? #336885
> I reverted this PR. Please implement your changes again on top of mine.

Sorry, I have missed yours. Will do !

**Comment by doronbehar** [comment]
> > Excuse me?? #336885
> > I reverted this PR. Please implement your changes again on top of mine.
> 
> Sorry, I have missed yours. Will do !

What happened here exactly?

**Comment by GaetanLepage** [comment]
> What happened here exactly?

dotlambda already had a PR ready for this update and mine got merged before his.
Hence, he preferred to revert this one and merge the original PR.

**Comment by doronbehar** [comment]
> > What happened here exactly?
> 
> dotlambda already had a PR ready for this update and mine got merged before his. Hence, he preferred to revert this one and merge the original PR.

That's what happens when there is no active maintainer for a package to coordinate such changes.
